sdk/spring-sdk/README.md

```diff
@@ -0,0 +1,232 @@
+# Kalix Spring SDK
+
+(Note: since the Spring SDK is very recent, there is not yet proper documentation so this document aims to serve as a general introduction guide.)
+
+The goal of the Spring SDK is to provide a code-first approach to developing Kalix applications so that a developer does not need to worry about learning protobuf upfront.
+
+This SDK offers a way to develop applications which should be familiar to someone who has worked with Spring applications by making use of annotations to assemble Kalix and its components.
+
+If you're new to Kalix and the different types of entities that exist in a Kalix app, consider reading [this section](https://docs.kalix.io/services/programming-model.html#_what_is_a_kalix_service) before continuing.  
+
+## Features
+
+As the Spring SDK is more recent than their gRPC-first counterparts, not all features are supported at this time. However, there's already a lot to play with as described next.
+
+### Value Entities
+
+To define a Value Entity, one will need to do 4 simple steps:
+1. create a class that extends `ValueEntity<S>`, where `S` is the state type this entity will store (what might usually be called a domain model class)
+2. make sure to annotate such class with `@Entity` and pass in an `entityKey` and an `entityType`  
+3. make use of Spring's RequestMapping annotation to define the routes to your entity
+4. declare the methods that will handle the requests and update the state of the entity
+
+```java
+// ...
+import kalix.springsdk.annotations.Entity;
+import kalix.javasdk.valueentity.ValueEntity;
+
+@Entity(entityKey = "id", entityType = "user") // <2>
+@RequestMapping("/user/{id}") // <3>
+public class UserEntity extends ValueEntity<User> { // <1>
+    
+    @PostMapping("/{email}/{name}") // <3>
+    public Effect<String> createOrUpdate(@PathVariable String email, @PathVariable String name) {
+        return effects()
+                .updateState(new User(email, name)) // <4>
+                .thenReply("Ok");
+    }
+}
+```
+From the above example, note:
+- the entityKey `id` needs to match a path parameter and such value needs to be unique per entity
+- the entityType `user` is common for all instances of this entity but should be unique across the different entity types
+- the API to update the state is the same as the Java one (and is provided by `effects()`)
+
+
+Below you can find a simple example of what the `User` model for the above entity could look like:
+```java
+public class User {
+  public String email;
+  public String name;
+  
+  @JsonCreator
+  public User(@JsonProperty("email") String email, @JsonProperty("name") String name) {
```

Comment:
I know that's never clear and sometimes we find situations that it doesn't work properly. But if someone start from our archetype, they should have everything to not need to defined `@JsonProperty`. Certainly not with explicitly name.

There is one situation where `@JsonProperty` is required. That's when the constructor has only one parameter.